### PR TITLE
LPD-32757 Implement endpoints for retrieving document data types

### PR DESCRIPTION
**Pull request by jkappler**
Implement endpoints for retrieving document types

# What is this trying to solve?

[Link to ticket](https://issues.liferay.com/browse/LPD-32757)

The changes add three new enpoints for retrieving document types. 

1. Retrieve the document types for a site
2. Retrieve the document types for a depot
3. Retrieve the document type by ID 

# How am I fixing it?

Add the required endpoints to the yaml file and implement them

# How can you verify that it works?

Run the integration tests, or go to /o/api and retrieve the document types by either ID, site or depot

# Why doesn't this include any test?

### CODE REVIEW

File: modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -6460,9 +6460,41 @@ paths:
                                 type: array
             tags: ["ContentTemplate"]
     "/asset-libraries/{assetLibraryId}/document-data-definition-types":
+        get:
```
Comment (Mikellorza) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml:6463]: Does it make sense to have also page and pageSize?

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentDataDefinitionTypeResourceTest.java [hunk 2]
```diff
@@ -92,13 +99,49 @@ public void testPostSiteDocumentDataDefinitionTypes() throws Exception {
...
+		throws Exception {
+
+		DocumentDataDefinitionType documentDataDefinitionType =
+			super.randomDocumentDataDefinitionType();
+
+		documentDataDefinitionType.setAvailableLanguages(
+			() -> LocaleUtil.toW3cLanguageIds(
+				new Locale[] {LocaleUtil.getDefault()}));
+		documentDataDefinitionType.setDataLayout(new DataLayout());
+		documentDataDefinitionType.setDataDefinitionFields(
```
Comment (Mikellorza) [inline modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentDataDefinitionTypeResourceTest.java:125]: Maybe setDataDefinitionFields should go before setDataLayout

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentDataDefinitionTypeResourceImpl.java [hunk 3]
```diff
@@ -34,16 +40,64 @@
...
+	extends BaseDocumentDataDefinitionTypeResourceImpl {
+
+	@Override
+	public Page<DocumentDataDefinitionType>
+			getAssetLibraryDocumentDataDefinitionTypesPage(Long assetLibraryId)
+		throws Exception {
+
+		return _getDocumentDataDefinitionTypesPage(
+			HashMapBuilder.put(
+				"createBatch",
```
Comment (Mikellorza) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentDataDefinitionTypeResourceImpl.java:57]: Does it make sense to have also [create](https://github.com/liferay/liferay-portal/blob/48d2ff809e9ff838012d6dd5fd48c19dc069cad6/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java#L177) and [get](https://github.com/liferay/liferay-portal/blob/48d2ff809e9ff838012d6dd5fd48c19dc069cad6/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java#L192) ?

Comment (Mikellorza) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentDataDefinitionTypeResourceImpl.java:57]: Also, does it make sense to have actions per item [here](https://github.com/liferay/liferay-portal/blob/2d163697d7a240a67561afcdaa7b5c1cccb036dc/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentDataDefinitionTypeResourceImpl.java#L209)? maybe only the get like https://github.com/liferay/liferay-portal/blob/48d2ff809e9ff838012d6dd5fd48c19dc069cad6/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java#L974

**Review by Mikellorza — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml (hunk 1 above)
Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml:6463]: Done

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentDataDefinitionTypeResourceTest.java (hunk 2 above)
Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentDataDefinitionTypeResourceTest.java:125]: Done

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentDataDefinitionTypeResourceImpl.java (hunk 3 above)
Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentDataDefinitionTypeResourceImpl.java:69]: Done

**Comment by jkappler** [comment]
@Mikellorza, changes applied. Thanks for the review! I also refactored the search to allow pagination as you mentioned ❤️

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:test:sf

**Review by Mikellorza — APPROVED** [review]

**Comment by Mikellorza** [comment]
LGTM!

**Comment by Mikellorza** [comment]
ci:forward

**Pull request closed without merging**